### DISCUSSION TITLE
How should I use the filter in the V2 version of the api

### DISCUSSION THREAD

**Discussion by Robert-Stackflow** [usememos/memos#3149] [Q&A] [answered]
As an example, when I access /api/v2/memos, I want to filter the list of public memos for a certain user. How should I use filter to achieve this function? I really don't know how to write the filter field

**Comment by lincolnthalles** [comment, marked as answer]
The filter syntax is a little annoying and it requires proper URI encoding.

See [Explore.tsx](https://github.com/usememos/memos/blob/release/0.21.0/web/src/pages/Explore.tsx#L35).

I've just updated a user script that shows a random memo button that also depends on this API. You can check it in this [gist](https://gist.github.com/lincolnthalles/957ee1b48634d4c41f5054526b17b5b0).

**Reply by Robert-Stackflow** [reply]
Thank you very much! You solved my problem!
